File: prepare.pl
#!/usr/bin/perl -w

#  prepare.pl
#  ApportablePlugin
#
#  Created by John Holdsworth on 04/05/2014.
#

use IO::File;
use strict;

my ($implementation) = @ARGV;
my ($projectMain) = split "\n", `find . -name 'main.m*' -print`;

system "chmod +w '$projectMain'";
IO::File->new( ">> $projectMain" )->print( <<CODE );

//#ifdef DEBUG
#import "$implementation"
//#endif
CODE

system "open '$projectMain'";

File: prepare.py
#!/usr/bin/python

#  prepare.py
#  ApportablePlugin
#
#  Created by John Holdsworth on 11/05/2014.
#

import sys
from subprocess import Popen, PIPE, call

implementation = sys.argv[1]

main_m = Popen(["find", ".", "-name", "main.m*", "-print"],
               stdout=PIPE).stdout.read().split( "\n" )[0]

call(["chmod", "+w", main_m])

open(main_m, "a").write("""
//#ifdef DEBUG
#import "%s"
//#endif
""" % implementation)

call(["open", main_m])
